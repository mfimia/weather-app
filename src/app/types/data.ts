export interface IWeather {
  date: string
  temperature: number
  condition: string
  iconUrl: string
  humidity?: number
  windSpeed?: number
}

export interface UIComponent {
  type: string
  text: string
}

export interface CityWeatherData {
  [cityName: string]: IWeather[]
}

export interface UIData {
  ui: UIComponent[]
}

export type IWeatherData = CityWeatherData & UIData

export interface IWeatherPayload {
  cityName: Cities
  cityData: IWeather[]
  ui: UIComponent[]
}

export enum Cities {
  Düsseldorf = 'Düsseldorf',
  Málaga = 'Málaga',
  Berlin = 'Berlin'
}
