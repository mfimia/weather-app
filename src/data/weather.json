{
  "Düsseldorf": [
    {
      "date": "current",
      "temperature": 15,
      "condition": "Cloudy",
      "humidity": 68,
      "windSpeed": 10,
      "iconUrl": "/cloudy.svg"
    },
    {
      "date": "Monday",
      "temperature": 14,
      "condition": "Cloudy",
      "iconUrl": "/cloudy.svg"
    },
    {
      "date": "Tuesday",
      "temperature": 15,
      "condition": "Partly Cloudy",
      "iconUrl": "/partly_cloudy.svg"
    },
    {
      "date": "Wednesday",
      "temperature": 16,
      "condition": "Sunny",
      "iconUrl": "/sunny.svg"
    }
  ],
  "Málaga": [
    {
      "date": "current",
      "temperature": 25,
      "condition": "Sunny",
      "humidity": 50,
      "windSpeed": 5,
      "iconUrl": "/sunny.svg"
    },
    {
      "date": "Monday",
      "temperature": 26,
      "condition": "Partly Cloudy",
      "iconUrl": "/partly_cloudy.svg"
    },
    {
      "date": "Tuesday",
      "temperature": 25,
      "condition": "Sunny",
      "iconUrl": "/sunny.svg"
    },
    {
      "date": "Wednesday",
      "temperature": 24,
      "condition": "Cloudy",
      "iconUrl": "/cloudy.svg"
    }
  ],
  "Berlin": [
    {
      "date": "current",
      "temperature": 13,
      "condition": "Rainy",
      "humidity": 75,
      "windSpeed": 12,
      "iconUrl": "/rainy.svg"
    },
    {
      "date": "Monday",
      "temperature": 12,
      "condition": "Rainy",
      "iconUrl": "/rainy.svg"
    },
    {
      "date": "Tuesday",
      "temperature": 13,
      "condition": "Cloudy",
      "iconUrl": "/cloudy.svg"
    },
    {
      "date": "Wednesday",
      "temperature": 14,
      "condition": "Partly Cloudy",
      "iconUrl": "/partly_cloudy.svg"
    }
  ],
  "ui": [
    {
      "type": "header",
      "text": "Weather for {cityName}"
    },
    {
      "type": "weatherInfo"
    },
    {
      "type": "daysForecast"
    }
  ]
}