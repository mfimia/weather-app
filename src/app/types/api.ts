import { IWeatherPayload } from './data'

export interface IApiResponse {
  code: number
  data: IWeatherPayload
}
