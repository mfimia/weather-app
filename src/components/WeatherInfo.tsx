import { IWeather } from '@/app/types/data'
import Image from 'next/image'
import React, { FC } from 'react'

interface IWeatherInfoProps {
  currentWeather: IWeather
}

const WeatherInfo: FC<IWeatherInfoProps> = ({ currentWeather }) => {
  return (
    <div data-testid="weatherInfo">
      <div className="flex items-center mb-4">
        <Image
          className="mr-4"
          src={currentWeather.iconUrl}
          alt={currentWeather.condition}
          width={64}
          height={64}
        />
        <div>
          <p className="text-xl">{`Temperature: ${currentWeather.temperature}Â°C`}</p>
          <p className="text-gray-600">{`Condition: ${currentWeather.condition}`}</p>
        </div>
      </div>
      <div className="mb-4">
        <p className="text-sm text-gray-600">{`Humidity: ${currentWeather.humidity}%`}</p>
        <progress
          value={currentWeather.humidity}
          max="100"
          className="w-full h-2 bg-blue-200"
        ></progress>
      </div>
      <p className="text-gray-600 mb-4">{`Wind Speed: ${currentWeather.windSpeed} km/h`}</p>
    </div>
  )
}

export default WeatherInfo
