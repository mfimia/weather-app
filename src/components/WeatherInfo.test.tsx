import { screen, render } from "@testing-library/react"
import { IWeather } from "@/app/types/data";
import WeatherInfo from "./WeatherInfo";

const mockCurrentWeather: IWeather = {
  date: "2023-10-06",
  temperature: 22,
  condition: "Sunny",
  iconUrl: "https://example.com/sunny-icon.png",
  humidity: 58,
  windSpeed: 10
};


describe('DayForecast', () => {
  it('should render correctly', () => {
    render(<WeatherInfo currentWeather={mockCurrentWeather} />)
    const weatherInfoElement = screen.getByTestId('weatherInfo')

    expect(weatherInfoElement).toBeVisible()
    expect(weatherInfoElement).toMatchSnapshot()
  })
})