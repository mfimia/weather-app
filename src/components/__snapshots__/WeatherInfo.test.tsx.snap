// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayForecast should render correctly 1`] = `
<div
  data-testid="weatherInfo"
>
  <div
    class="flex items-center mb-4"
  >
    <img
      alt="Sunny"
      class="mr-4"
      data-nimg="1"
      decoding="async"
      height="64"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fexample.com%2Fsunny-icon.png&w=128&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fsunny-icon.png&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fsunny-icon.png&w=128&q=75 2x"
      style="color: transparent;"
      width="64"
    />
    <div>
      <p
        class="text-xl"
      >
        Temperature: 22Â°C
      </p>
      <p
        class="text-gray-600"
      >
        Condition: Sunny
      </p>
    </div>
  </div>
  <div
    class="mb-4"
  >
    <p
      class="text-sm text-gray-600"
    >
      Humidity: 58%
    </p>
    <progress
      class="w-full h-2 bg-blue-200"
      max="100"
      value="58"
    />
  </div>
  <p
    class="text-gray-600 mb-4"
  >
    Wind Speed: 10 km/h
  </p>
</div>
`;
